### PR TITLE
For review only - Initial GemFire binder

### LINKED ISSUE #214: Support consumer group in Gemfire binder
The concept of consumer groups was recently introduced to the binder SPI. This needs to be implemented in the Gemfire binder.

### PR DESCRIPTION
**Pull request by pperalta**
* Resolves #214
* Consumer group support via replicated region (cc @markfisher)
* Message processing via [GemFire async event queue](http://gemfire.docs.pivotal.io/docs-gemfire/developing/events/implementing_write_behind_event_handler.html) - could potentially be moved into SI (cc @garyrussell @mbogoevici)

### CODE REVIEW

File: spring-cloud-stream-binders/spring-cloud-stream-binder-gemfire/src/main/java/org/springframework/cloud/stream/binder/gemfire/GemfireMessageChannelBinder.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		// todo: obtain properties from the config classes
+		Properties properties = new Properties();
+		properties.put("locators", "localhost[7777]");
+		properties.put("log-level", "warning");
+		properties.put("mcast-port", "0");
+		this.cache = new CacheFactory(properties).create();
+		RegionFactory<String, Set<String>> regionFactory = this.cache.createRegionFactory(RegionShortcut.REPLICATE);
+		this.consumerGroupsRegion = regionFactory.create(CONSUMER_GROUPS_REGION);
```
Comment (garyrussell) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-gemfire/src/main/java/org/springframework/cloud/stream/binder/gemfire/GemfireMessageChannelBinder.java]: This region name probably needs to be configurable too.

**Comment by garyrussell** [comment]
LGTM after a quick scan - I concur we should consider adding an `AsyncEventListener` - based inbound channel adapter to SI.

File: spring-cloud-stream-binders/spring-cloud-stream-binder-gemfire/pom.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-core</artifactId>
+			<optional>true</optional>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-gemfire</artifactId>
+			<version>1.7.0.RELEASE</version>
```
Comment (mbogoevici) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-gemfire/pom.xml]: Cosmetic: should move versions into properties for easier management.

**Comment by pperalta** [comment]
Closed in favor of https://github.com/spring-cloud/spring-cloud-stream-binder-gemfire/

**Pull request closed without merging**